packages/utils/telemetry-utils/src/staging.ts

```diff
@@ -0,0 +1,43 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ILoggingError } from "@fluidframework/common-definitions";
+
+// ///////////////////////////////////////////////////////////////////////// //
+//        THIS CODE TO BE MOVED TO COMMON-DEFINITIONS AND COMMON-UTILS       //
+// ///////////////////////////////////////////////////////////////////////// //
+
+export interface IFluidErrorBase extends Readonly<Error> {
+    readonly errorType: string;
+    readonly fluidErrorCode: string;
+}
+
+export function isFluidError(e: any): e is IFluidErrorBase {
+    return typeof e.fluidErrorCode === "string";
+}
+
+/** type guard to ensure it has an errorType e.g. via IErrorBase */
+export const hasErrorType = (error: any): error is { errorType: string } => {
+    return (typeof error?.errorType === "string");
+};
+
+/** type guard for ILoggingError interface */
+export const isILoggingError = (x: any): x is ILoggingError => typeof x?.getTelemetryProperties === "function";
+
+export const isErrorLike = (x: any): x is Error =>
+    typeof(x?.message) === "string" &&
+    typeof(x?.name) === "string" &&
+    (x?.stack === undefined || typeof(x?.stack) === "string");
+
+/** This type lets you mixin T onto an existing object */
+export type Builder<T> = {
```

Comment:
this won't always be safe, if the property is a getter, or the object is frozen setting will fail at runtime. can you make this more specific, as i think having it super general could get people in trouble.

Reply:
you only use this with IFluidErrorBase. May just have a writable version of that, and `IFluidErrorBase = Readonly<IFluidErrorBaseBuilder>`